atc/api/accessor/accessor.go

```diff
@@ -14,52 +17,158 @@ type Access interface {
 	IsAdmin() bool
 	IsSystem() bool
 	TeamNames() []string
-	CSRFToken() string
 	UserName() string
+	UserInfo() UserInfo
+}
+
+type UserInfo struct {
+	Sub      string
+	Name     string
+	UserID   string
+	UserName string
+	Email    string
+	IsAdmin  bool
+	IsSystem bool
+	Teams    map[string][]string
+}
+
+type Verification struct {
+	HasToken     bool
+	IsTokenValid bool
+	Claims       map[string]interface{}
 }
 
 type access struct {
-	*jwt.Token
-	action        string
-	actionRoleMap ActionRoleMap
+	verification      Verification
+	requiredRole      string
+	systemClaimKey    string
+	systemClaimValues []string
+	teams             []db.Team
+}
+
+func NewAccessor(
+	verification Verification,
+	requiredRole string,
+	systemClaimKey string,
+	systemClaimValues []string,
+	teams []db.Team,
+) *access {
+	return &access{
+		verification:      verification,
+		requiredRole:      requiredRole,
+		systemClaimKey:    systemClaimKey,
+		systemClaimValues: systemClaimValues,
+		teams:             teams,
+	}
 }
 
 func (a *access) HasToken() bool {
-	return a.Token != nil
+	return a.verification.HasToken
 }
 
 func (a *access) IsAuthenticated() bool {
-	return a.HasToken() && a.Token.Valid
+	return a.verification.IsTokenValid
 }
 
-func (a *access) Claims() jwt.MapClaims {
-	if a.IsAuthenticated() {
-		if claims, ok := a.Token.Claims.(jwt.MapClaims); ok {
-			return claims
+func (a *access) IsAuthorized(teamName string) bool {
+
+	if a.IsAdmin() {
+		return true
+	}
+
+	for _, team := range a.TeamNames() {
+		if team == teamName {
+			return true
 		}
 	}
-	return jwt.MapClaims{}
+
+	return false
 }
 
-func (a *access) IsAuthorized(team string) bool {
-	if a.IsAdmin() {
-		return true
+func (a *access) TeamNames() []string {
+
+	teamNames := []string{}
+
+	isAdmin := a.IsAdmin()
```

Comment:
could this be cached on the access object ?
When `IsAuthorized` is called and user isn't an admin, this function is called repeatedly.